modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/exception/NoSuchDataDefinitionException.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.data.engine.rest.internal.exception;
```

Comment:
this exception should be in the -api module. Please, search for similar uses.
you should create this class in `data-engine-api` `com.liferay.data.engine.exception`

Reply:
Ok doing it.